src/managed_blmo.jl

```diff
@@ -69,6 +69,111 @@ function compute_extreme_point(blmo::ManagedBoundedLMO, d; kwargs...)
     return v
 end
 
+#================================================================================================================#
+function is_decomposition_invariant_oracle(blmo::ManagedBoundedLMO)
+    sblmo = blmo.simple_lmo
+    
+    if typeof(sblmo) == CubeSimpleBLMO
+        lbs = sblmo.lower_bounds
+        ubs = sblmo.upper_bounds
+        indicator1 = [0.0, 1.0]
+        indicator2 = [1.0, 0.0]
+        distinct_lbs = unique(lbs)
+        distinct_ubs = unique(ubs)
+        if !issubset(distinct_lbs, indicator1) && !issubset(distinct_lbs, indicator2)
+            return false
+        end
+        if !issubset(distinct_ubs, indicator1) && !issubset(distinct_ubs, indicator2)
+            return false
+        end
+    end
+
+    if typeof(sblmo) == ProbabilitySimplexSimpleBLMO
+        if !(sblmo.N == 1)
+            return false
+        end
+    end
+
+    if typeof(sblmo) == UnitSimplexSimpleBLMO
+        if !(sblmo.N == 1)
+            return false
+        end
+    end
+
+    return true
+end
+
+# Provide FrankWolfe.compute_inface_extreme_point
+function compute_inface_extreme_point(blmo::ManagedBoundedLMO, direction, x; kwargs...)
+    time_ref = Dates.now()
+    sblmo = blmo.simple_lmo
+
+    #CubeSimpleBLMO{T}(lower_bounds, upper_bounds)
+    
+    if typeof(sblmo) == CubeSimpleBLMO
+        a = bounded_compute_inface_extreme_point(
+                blmo.simple_lmo,
+                direction,
+                x,
+                blmo.lower_bounds,
+                blmo.upper_bounds,
+                blmo.int_vars,
+                )
+        blmo.solving_time = float(Dates.value(Dates.now() - time_ref))
+        return a
+    end
+    
+    #ProbablitySimplexSimpleBLMO(N)
+    #Scaled Probability Simplex: ∑ x = 1.
+
+    if typeof(sblmo) == ProbabilitySimplexSimpleBLMO
+        lmo = FrankWolfe.ProbabilitySimplexOracle(1.0)
+        a = FrankWolfe.compute_inface_extreme_point(lmo, direction, x,)
```

Comment:
Careful, you will probably have to adapt the function in the Boscia context because the probability simplex will see additional bounds latest after the first branching!